ntirpc/rpc/gss_internal.h

```diff
@@ -119,6 +119,9 @@ unref_svc_rpc_gss_data(struct svc_rpc_gss_data *gd)
 	/* if refcnt is 0, gd is not reachable */
 	if (unlikely(atomic_dec_uint32_t(&gd->refcnt) == 0)) {
 		svcauth_gss_destroy(gd->auth);
+		/* gd was freed.  We can't unlock the lock.  But, it's safe to
+		 * destroy a lock if you own it.  */
+		return;
```

Comment:
Not sure if I understand the comment "But, it's safe to destroy a lock if you own it.".  In fact, svcauth_gss_destroy() calls unlock and destroy over gd_lock. This is bad idiom, why we need lock around svcauth_gss_destroy() ? Can we completely remove as the atomic refcount should ensure that there is only one thread calling svcauth_gss_destroy() ?

Reply:
We call svcauth_gss_destroy() directly from unref_svc_rpc_gss_data() with refcounts but we also pass svcauth_gss_destroy() as a function pointer in svc_auth_gss_ops structure. Who calls it and when? Can svcauth_gss_destroy() get called with active refcount on gd?

Reply:
yes, now that there are assured to be no refs--and no more can be acquired--it's safe to destroy it